### PR TITLE
Ensure `plot_emission_intensity` with input data having `label` column, outputs custom colour scale with expected order

### LINKED ISSUE #535: Adding column `label` to input of `plot_emission_intensity` in conjunction with `scale_colour_r2dii`
When the user is trying to use the `label` column to data to change the appearance of the labels but also uses `factor()` to change the order (and colours) of the metrics the two don't align and the plot ends up having wrong labels assigned to colours.

### PR DESCRIPTION
**Pull request by jdhoffa**
Closes #535

### CODE REVIEW

**Comment by jdhoffa** [comment]
Note: My test isn't working as expected 😂

**Comment by jdhoffa** [comment]
@MonikaFu what I did here was, in the `add_label` function, I added a condition that checks if the `metric(data)` column to which the `label` column corresponds is factored. If it is, it then factors the `label` column with the same order.

**Review by MonikaFu — CHANGES_REQUESTED** [review]

File: tests/testthat/test-plot_emission_intensity.R [hunk 1]
```diff
@@ -87,3 +87,80 @@ test_that("throws expected warning about API change", {
     class = "warning"
   )
 })
+
+test_that("with data with `label` column, outputs custom colour scale with expected order (#535)", {
```
[the remark this answers is not in the document]
Comment (jdhoffa) [reply tests/testthat/test-plot_emission_intensity.R:91]: Hmm, I thought "outputs custom colour scale" implied that it was a custom colour scale? 

But happy to make it more clear if you think my wording was confusing

[the remark this answers is not in the document]
Comment (jdhoffa) [reply tests/testthat/test-plot_emission_intensity.R:91]: Do you like
"with data with `label` column and with `scale_colour_r2dii`, outputs expected labels and colours"

File: R/utils.R [hunk 2]
```diff
@@ -224,6 +224,20 @@ span_5yr <- function(data) {
 
 add_label_if_missing <- function(data) {
```
[the remark this answers is not in the document]
Comment (jdhoffa) [reply R/utils.R:225]: Good point!

[the remark this answers is not in the document]
Comment (jdhoffa) [reply R/utils.R:225]: `add_label_and_factor_if_necessary`? lol

File: tests/testthat/test-plot_emission_intensity.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (MonikaFu) [reply tests/testthat/test-plot_emission_intensity.R:91]: Yes!

File: R/utils.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (MonikaFu) [reply R/utils.R]: I'm happy with how you resolved this! however, I see all checks failing. I guess something to look into.

File: tests/testthat/test-plot_emission_intensity.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdhoffa) [reply tests/testthat/test-plot_emission_intensity.R]: Should be resolved :-)

File: R/utils.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (jdhoffa) [reply R/utils.R]: Yup, lol hasty work. 

I was returning `data$label` instead of `data`. 

Fixed :-)

**Comment by jdhoffa** [comment]
@MonikaFu I'm inclined to NOT update the `NEWS.md` here, since we are fixing a bug that was introduced only in dev, so no user-facing changes should be observed since the previous release, do you agree?

**Comment by jdhoffa** [comment]
The old release failing checks to be handled here: https://github.com/RMI-PACTA/r2dii.plot/issues/540

**Pull request merged**